pkg/registry/controller/rest.go

```diff
@@ -47,14 +48,25 @@ func (rcStrategy) NamespaceScoped() bool {
 func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 	controller := obj.(*api.ReplicationController)
 	controller.Status = api.ReplicationControllerStatus{}
+
+	// We could allow users to initialize the sequenceNumbers, but since
+	// we don't allow them to update it, doing so would be non-intuitive.
+	controller.Spec.SequenceNumber = 0
+	controller.Status.SequenceNumber = 0
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (rcStrategy) PrepareForUpdate(obj, old runtime.Object) {
-	// TODO: once RC has a status sub-resource we can enable this.
-	//newController := obj.(*api.ReplicationController)
-	//oldController := old.(*api.ReplicationController)
-	//newController.Status = oldController.Status
+	newController := obj.(*api.ReplicationController)
+	oldController := old.(*api.ReplicationController)
+
+	// Any changes to the spec increment the sequence number, any changes to the
+	// status should reflect the sequence number of the corresponding spec. We push
+	// the burden of managing the status onto the clients to avoid confusion when
+	// we have concurrent updates.
```

Comment:
Suggest s/to avoid confusion when we have concurrent updates./because we can't (in general) know here what version of spec the writer of the status has seen. It may seem like we can at first--since obj contains the spec--but in the future we will probably make status its own object, and even if we don't, writes may be the result of a read-update-write loop, so the contents of spec may not actually be the spec that the controller has actually "seen"./